### ISSUE TITLE
Command: no helpful error message when something is wrong with constructor

### ISSUE THREAD

**Issue by flip111** [symfony/symfony#16496]
I have a Command class with nothing really special in it. The configure method names the command `user:create`, when i make the constructor like this

``` php
  public function __construct() {
    parent::__construct();
  }
```

Everything is fine.

When i make it like this

``` php
  public function __construct($name) {
    parent::__construct($name);
  }
```

I get error

```
[InvalidArgumentException]
There are no commands defined in the "user" namespace.
```

Expected to tell me something about the constructor function

**Comment by iltar** [comment]
@flip111 my guess is that it goes wrong here: [&& !$r->getConstructor()->getNumberOfRequiredParameters()](https://github.com/symfony/symfony/blob/df9d0821b0a1f90eaed2ac20dbb041fda65febd1/src/Symfony/Component/HttpKernel/Bundle/Bundle.php#L200), this code is from `Bundle.php` and is used to determine whether or not the command should be added. On the next line it will add the command to the application and instantiate it without arguments. I think this case should not be skipped but throw an exception instead.

**Comment by xabbuh** [comment]
@iltar Well, the command could still be registered with the `console.command` service tag. So throwing an exception here is not a nice solution imho.

**Comment by iltar** [comment]
@xabbuh that's only with Command as a Service, it will only hit this case if it's not registered as service. If it _is_ registered, it will continue the iteration before hitting this part. Adding Command as a Service is done [here](https://github.com/symfony/symfony/blob/3909460c04e6b13858f2fa3b4e41ec68fdebf0a2/src/Symfony/Bundle/FrameworkBundle/Console/Application.php#L101-L116).

**Comment by jvasseur** [comment]
You could have a command that is conditionally added as a service by the bundle extension depending on the config.

You should probably override the registerCommands method in this case but that's still a use case for the current way things works.

**Comment by taueres** [comment]
I had this same problem today. I think Symfony should at least emit a Warning and inform the user that something is wrong.
Do you think `trigger_error("Wrong constructor for command X", E_USER_WARNING);` would be a possible solution?

**Comment by xabbuh** [comment]
Throwing an exception can be considered a BC break as the application did not break before. We will need to check if we can just output a warning message in the application.

**Comment by HeahDude** [comment]
It the warning is triggered silently it will only appear in the logs (so debug toolbar) as silenced error.

**Comment by HeahDude** [comment]
If not it depends on the dev's environment config.

**Comment by chalasr** [comment]
This could be solved easily on 3.4+ as errors occurring during the command registration process are now caught and displayed as warnings, so we could safely throw in this case. 
But, we just deprecated the convention based registration and the default config already registers commands as services since 3.3 thanks to the PSR-4 loader, so you already get a proper error.
I would close as won't fix, not worth it.

**Comment by chalasr** [comment]
Closing as explained.
